### PR TITLE
feat(#1588): WYSIWYG Mac host chrome — PR2 panels

### LINKED ISSUE #1588: WYSIWYG slice 4 PR2 — native panels: inspector, palette, drag-in, semantic paste
Part of #1221 (epic), sub-issue of #1225 (slice 4 — Mac host chrome). PR2 of 3 per `docs/superpowers/specs/2026-08-10-wysiwyg-mac-host-chrome-design.md` and `docs/superpowers/plans/2026-08-10-wysiwyg-mac-host-plumbing.md`. PR1 (plumbing) shipped in #1400; this is the remainder that #1225 was mistakenly closed before landing (see the reopening comment on #1225 for how).

- Native inspector for typed block props using system controls (not the interim palette from PR1).
- Native palette as a source list, replacing PR1's interim block palette.
- Finder/Photos drag-in → asset ingestion + image block insertion.
- Semantic paste: rich text → blocks + honest runs (⇧⌥⌘V for "paste and match style" semantics).
- Block copy puts HTML + plain text on the pasteboard.

Spec §8.3–8.4.

### PR DESCRIPTION
**Pull request by davidwkeith**
Part of #1225 (slice 4 — Mac host chrome), part of epic #1221. This is PR2 of 3 per the design spec's PR breakdown ([spec](docs/superpowers/specs/2026-08-10-wysiwyg-mac-host-chrome-design.md), [plan](docs/superpowers/plans/2026-08-20-wysiwyg-mac-host-panels.md)); PR1 (plumbing) shipped as #1400, PR3 (polish/keyboard/VoiceOver/text-services) follows separately, so this PR intentionally does not close #1588.

## Summary

- Native inspector for typed block props (steppers, color wells, pop-up buttons) routed into the existing unified window inspector, backed by a real canvas-selection round-trip (JS click-to-select → native `selectedBlockId`) and a parent-lookup helper generalizing block duplicate/delete to nested blocks.
- Native block palette (source list) with in-canvas drag-reorder (`DragReorderController` wired into the JS mount path, with a visible drag handle and a live drop-target indicator) and cross-boundary drag-drop insert from the palette into the canvas.
- Finder/Photos image drag-in → asset ingestion (magic-byte sniffed into `public/images/`) → image block insertion in one gesture. AI alt-text stays a stubbed empty string (deferred to slice 6, #1227).
- Semantic paste (RTF/HTML → blocks with honest inline runs) and ⇧⌥⌘V "Paste and Match Style"; block copy puts real HTML + plain text on the pasteboard, wired via `.onCopyCommand` (not a Commands-menu item, to avoid duplicating the system's default Edit ▸ Copy — same fix already applied to Delete via #989/#1423).
- Document conventions: an edited-dot tied to uncommitted ops, the window title/proxy icon pointing at the specific page being edited, a native ⌘F find bar (`WKWebView.find`), and a customizable toolbar item for the palette panel.
- A whole-branch review after all 20 plan tasks found and fixed 4 cross-task issues no single task's diff-scoped review could see on its own: an attribute-injection gap in the HTML clipboard writer's escaping, a JS/Swift root-slot naming mismatch (`"default"` vs `"main"`) on the drop-insertion path, four silent-failure points in the drop pipeline (now logged via `LogCenter`), and an invisible drag-reorder handle with a discarded drop indicator (now visible, with real feedback and scroll/resize handling).

## Known limitations (by design, not oversight)

- **No real backend yet.** `StubWYSIWYGHostTransport` still applies ops to an in-memory model only — no real source write, no git commit. Unchanged from PR1; waits on #1222.
- **`DragReorderController`'s in-canvas reorder path** (`#handleUp` in `JS/wysiwyg-engine/src/drag-drop.ts`) still uses `slot: "default"` for `moveBlock`, the same JS/Swift root-slot naming mismatch fixed on the drop-insertion path during the final review — currently inert (no real backend consumes `slot` at the root yet), left for a follow-up once #1222 lands, since the reorder and drop-insertion defaults need to change together.
- **Semantic paste's rich-run mapping (bold/italic/link detection) has no production call site yet** — `WYSIWYGPasteCommands` only ever calls the mapper with `plainTextOnly: true` (Paste and Match Style). The rich-run code path is fully implemented and tested but not user-reachable via a gesture in this PR.
- **Multi-frame breakpoint support stays deferred.** PR1's mount path is still a single document/single `WysiwygEngine`; `BreakpointCanvas` remains unwired to the native host, so the design doc's known limitations (b)/(c) around per-frame `DragReorderController` instances and frame-local coordinate translation are addressed only for the single-document case this slice actually mounts.
- **Paste and Match Style always inserts at the page root**, not adjacent to the current selection — inconsistent with `duplicateSelectedBlock`/the palette drop (which do honor position), but matches the plan's own snippet; worth revisiting.
- A handful of test-coverage gaps and narrow edge cases were found, reviewed, and explicitly deferred rather than fixed blind (e.g. find-bar chevron accessibility labels, CRLF paragraph splitting in the paste mapper, two adjacent `.listStyle(.sidebar)` panels when both the Navigator and the palette are open) — see individual commit messages and the plan's ledger for specifics.

## Paired PR check

- [x] This change is **self-contained** to `Anglesite/Anglesite`.
- [ ] This change **needs a paired PR** in [`Anglesite/anglesite-skills`](https://github.com/Anglesite/anglesite-skills) (MCP sidecar server).

## Test plan

- [x] `swift test --package-path .` — 659/659 tests passed, 83 suites, all green.
- [x] `scripts/build-app.sh -project Anglesite.xcodeproj -scheme Anglesite -configuration Debug build` — BUILD SUCCEEDED.
- [x] `cd JS/wysiwyg-engine && npm run lint && npm run typecheck && npm test` — 123/123 passing, lint and typecheck clean.
- [ ] Manual smoke: not performed in this environment (no interactive GUI automation available for a live WKWebView session) — recommend a manual pass before merge covering: toggling Site ▸ Edit Page (titlebar title/proxy icon/edited-dot), the inspector reacting to canvas block selection, dragging a palette row onto the canvas, reordering a block via its drag handle, dragging an image from Finder onto the canvas, ⇧⌥⌘V pasting styled text, ⌘C copying a block into another app, and ⌘F on the canvas.

🤖 Generated with Subagent-Driven Development (superpowers:subagent-driven-development) — 20 tasks, each independently implemented and reviewed (several after a fix round), plus a final whole-branch review and one fix wave. Full task-by-task history is in the commit log.

### CODE REVIEW

File: JS/wysiwyg-engine/src/host/mount.ts [hunk 1]
```diff
@@ -42,6 +260,12 @@ window.__anglesiteWysiwygMount = {
     // `HostTransport` and `QualityGateTransport` (#1226 Task 12), so one object owns the whole
     // `window.__anglesiteWysiwygHost` bridge.
     window.__anglesiteWysiwygQualityGates = new QualityGateChips(engine, transport);
+    disposeSelection = wireSelection(engine);
+    dropIndicator = renderDropIndicator();
+    // Real callback, not a no-op: `DragReorderController` already computes the live drop target on
+    // every `pointermove` and hands it to `onIndicator` precisely so a host can draw it.
+    dragReorder = new DragReorderController(engine, (target) => dropIndicator?.update(target), document);
```
Comment (davidwkeith) [inline JS/wysiwyg-engine/src/host/mount.ts]: **Bug: in-canvas drag-reorder always computes drop index 0.**

`DragReorderController` is constructed here with `document` as its container, but `computeDropTarget` (drag-drop.ts) queries `container.children` to find the block-id-bearing siblings to measure against. `document.children` is just `[<html>]` — no element there carries `BLOCK_ID_ATTR`, so `rects` is always empty and `nearestIndexInSlot` unconditionally returns `0`.

Every in-canvas drag-reorder gesture will therefore submit `moveBlock` with the drop target computed as index 0, regardless of where the block was visually dropped — the block always jumps to the top of the page.

Compare with `dropTargetAt` a few lines below (and `renderDropIndicator` above), which correctly query `document.body.children`. This constructor call should likely pass `document.body` instead of `document` as the container (note `startDrag`'s separate `doc` parameter still needs to stay `document` for the pointer-event listeners — only the geometry container is wrong here).

The one new test covering this path (`mount-drag.test.ts`) only has a single block on the page, so index 0 is coincidentally correct there and doesn't catch this.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: JS/wysiwyg-engine/src/host/mount.ts [hunk 2]
```diff
@@ -2,22 +2,240 @@ import { WysiwygEngine } from "../engine.js";
...
+    }
+    handle.style.display = "block";
+    handle.style.left = `${rect.x}px`;
+    handle.style.top = `${rect.y}px`;
+  };
+
+  const onPointerDown = (event: PointerEvent) => {
+    const blockId = engine.selection.current;
+    if (!blockId) return;
+    event.preventDefault();
```
Comment (davidwkeith) [inline JS/wysiwyg-engine/src/host/mount.ts]: Minor: this `pointerdown` handler doesn't check `event.button`, so a right-click (or trackpad secondary-click) on the drag handle also starts a reorder drag via `dragReorder.startDrag(blockId)`. `preventDefault()` here doesn't stop the separate `contextmenu` event, so the native context menu opens while a drag is considered in-progress; since the ensuing menu interaction consumes the pointer, no `pointerup` reaches `DragReorderController`'s listeners, and it's left believing a drag is still active until the next real drag or `dispose()`.

Worth gating on `event.button === 0` (primary button only), consistent with how most drag-initiation code treats non-primary buttons.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by davidwkeith — COMMENTED** [review]
Reviewed the diff (37 files, +4481/-70). The overall structure holds up well — the native↔JS selection round-trip, the nested-block `locate()` generalization, HTML-clipboard escaping, image magic-byte sniffing, and the `NATIVE_ROOT_SLOT`/`"main"` vs `"default"` handling all look correct and are consistent with their own tests.

Found one functional bug in the headline in-canvas drag-reorder feature (see inline comment on `mount.ts`): `DragReorderController` is constructed with `document` instead of `document.body` as its geometry container, so `computeDropTarget` never finds any block-id-bearing children and every reorder drop resolves to index 0 — reordering always moves the dragged block to the top of the page instead of where it was dropped. The single new drag test doesn't catch it because it only has one block on the page (index 0 is coincidentally correct there). This should be fixed before merge.

Also flagged a smaller edge case: the drag handle's `pointerdown` doesn't check `event.button`, so right-click on the handle can leave `DragReorderController` in a stuck "dragging" state until the next real drag.

Neither issue is mentioned in the PR's own "Known limitations" list, so they look like genuine gaps rather than already-flagged/deferred items. Everything else — the four fixes called out from the whole-branch review, `WindowEditedStateBridge`, `SiteWindowModel`'s stale-selection clearing — checked out.

---
_Generated by [Claude Code](https://claude.ai/code)_
